`verdi process watch` error

There seems to be a problem with the command `verdi process watch` where the following exception was reached:

```
07/04/2019 01:08:40 PM <19245> kiwipy.rmq.communicator: [ERROR] Exception in broadcast receiver
Traceback (most recent call last):
  File "/home/bonan/miniconda3/envs/aiida-1.0-main/lib/python2.7/site-packages/kiwipy/rmq/communicator.py", line 237, in _on_broadcast
    msg[messages.BroadcastMessage.CORRELATION_ID])
  File "/home/bonan/miniconda3/envs/aiida-1.0-main/lib/python2.7/site-packages/tornado/gen.py", line 1055, in run
    value = future.result()
  File "/home/bonan/miniconda3/envs/aiida-1.0-main/lib/python2.7/site-packages/tornado/concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "/home/bonan/miniconda3/envs/aiida-1.0-main/lib/python2.7/site-packages/tornado/gen.py", line 292, in wrapper
    result = func(*args, **kwargs)
  File "/home/bonan/miniconda3/envs/aiida-1.0-main/lib/python2.7/site-packages/kiwipy/filters.py", line 33, in __call__
    self._subscriber(communicator, body, sender, subject, correlation_id)
TypeError: _print() takes exactly 4 arguments (5 given)
```
The `_print` function takes 4 arguments:

https://github.com/aiidateam/aiida_core/blob/e232f946e2b5c1c55c8f8b2c903f05355fe9deea/aiida/cmdline/commands/cmd_process.py#L262-L273

But the callback gives 5 arguments in `kiwipy` here:

https://github.com/aiidateam/kiwipy/blob/f0b3ea3c6cd1ec9653586ee5ca5bdb6dcc84e565/kiwipy/filters.py#L9-L33

The `communicator` argument was not included in the `_print` function.

BTW I am not quite sure what exactly this command does and it appears to hang most of the time....

--- a/aiida/cmdline/commands/cmd_process.py
+++ b/aiida/cmdline/commands/cmd_process.py
@@ -269,13 +269,21 @@ def process_play(processes, timeout, wait):
 @decorators.only_if_daemon_running(echo.echo_warning, 'daemon is not running, so process may not be reachable')
 def process_watch(processes):
     """Watch the state transitions for a process."""
+    from time import sleep
     from kiwipy import BroadcastFilter
-    import concurrent.futures
 
-    def _print(body, sender, subject, correlation_id):
-        echo.echo('pk={}, subject={}, body={}, correlation_id={}'.format(sender, subject, body, correlation_id))
+    def _print(communicator, body, sender, subject, correlation_id):  # pylint: disable=unused-argument
+        """Format the incoming broadcast data into a message and echo it to stdout."""
+        if body is None:
+            body = 'No message specified'
+
+        if correlation_id is None:
+            correlation_id = '--'
+
+        echo.echo('Process<{}> [{}|{}]: {}'.format(sender, subject, correlation_id, body))
 
     communicator = get_manager().get_communicator()
+    echo.echo_info('watching for broadcasted messages, press CTRL+C to stop...')
 
     for process in processes:
 
@@ -286,9 +294,12 @@ def process_watch(processes):
         communicator.add_broadcast_subscriber(BroadcastFilter(_print, sender=process.pk))
 
     try:
-        # Block this thread indefinitely
-        concurrent.futures.Future().result()
+        # Block this thread indefinitely until interrupt
+        while True:
+            sleep(2)
     except (SystemExit, KeyboardInterrupt):
+        echo.echo('')  # add a new line after the interrupt character
+        echo.echo_info('received interrupt, exiting...')
         try:
             communicator.stop()
         except RuntimeError:

